app/views/QRScan.js

```diff
@@ -0,0 +1,291 @@
+      LocationServices.saveLocation({
+        latitude: Number(latitude),
+        longitude: Number(longitude),
+        time: Date.now(),
+      });
+      this.setState({ currentState: StateEnum.SCAN_SUCCESS });
+    } else {
+      this.setState({ currentState: StateEnum.SCAN_FAIL });
+    }
+  }
+
+  isValidCoordinates(latitude, longitude) {
+    const isValid = num => {
+      const regex = /^(-?\d+\.?\d*|\.\d+)$/;
+      return typeof num === 'string' && regex.test(num);
+    };
+    return isValid(latitude) && isValid(longitude);
+  }
+
+  handleBackPress = () => {
+    this.props.navigation.goBack();
+    return true;
+  };
+
+  getBackground() {
+    return BackgroundImage;
+  }
+
+  getMainText() {
+    switch (this.state.currentState) {
+      case StateEnum.SCAN_SUCCESS:
+        return (
+          <Typography style={styles.mainTextBelow}>
+            {languages.t('label.qr_successful_title')}
+          </Typography>
+        );
+      case StateEnum.SCAN_FAIL:
+        return (
+          <Typography style={styles.mainTextBelow}>
+            {languages.t('label.qr_fail_title')}
+          </Typography>
+        );
+    }
+  }
+
+  getSubText() {
+    switch (this.state.currentState) {
+      case StateEnum.SCAN_SUCCESS:
+        return languages.t('label.qr_successful_subtitle');
+      case StateEnum.SCAN_FAIL:
+        return languages.t('label.qr_fail_subtitle');
+    }
+  }
+
+  render() {
+    return (
+      <ImageBackground
+        source={this.getBackground()}
+        style={styles.backgroundImage}>
+        <StatusBar
+          barStyle='light-content'
+          backgroundColor='transparent'
+          translucent
+        />
+        {(this.state.currentState === StateEnum.SCAN_SUCCESS ||
+          this.state.currentState === StateEnum.SCAN_FAIL) && (
+          <View style={styles.iconContainer}>
+            <StateIcon size={height} status={this.state.currentState} />
+          </View>
+        )}
+        {(this.state.currentState === StateEnum.SCAN_SUCCESS ||
+          this.state.currentState === StateEnum.SCAN_FAIL) && (
+          <View style={styles.mainContainer}>
+            <View style={styles.content}>
+              {this.getMainText()}
+              <Typography style={styles.subheaderText}>
+                {this.getSubText()}
+              </Typography>
+              <View style={styles.buttonContainer}>
+                <Button
+                  label={languages.t('label.qr_exit')}
+                  onPress={() => {
+                    this.props.navigation.navigate('Main', {});
+                  }}
+                />
+              </View>
+            </View>
+          </View>
+        )}
+        {this.state.currentState === StateEnum.SCAN_IN_PROGRESS && (
+          <View style={styles.qrScanContainer}>
+            <QRCodeScanner
+              onRead={this.onScanSuccess.bind(this)}
+              flashMode={RNCamera.Constants.FlashMode.torch}
```

Comment:
I would suggest having flash off by default. Can think of use-case where flashlight might not be a good idea.

Reply:
Future ux improvement can add the ability to toggle the flashlight within the screen.